api/src/exceptions/database/translate.ts

```diff
@@ -38,7 +39,8 @@ export async function translateDatabaseError(error: SQLError): Promise<any> {
 			break;
 	}
 
-	const hookError = await emitter.emitAsync('database.error', defaultError, { client });
+	const hookResult = await emitter.emitAsync('database.error', defaultError, { client });
+	const hookError = Array.isArray(hookResult) ? last(compact(hookResult)) : hookResult;
```

Comment:
Should there ever be a reason to prevent the error from being thrown altogether 🤔

Reply:
Altering the error that's thrown is definitely nice to have, just wondering if there's a use case for preventing it from throwing the process completely

Reply:
(We can keep discussing this outside of this PR 👍🏻 )